Mathlib/Combinatorics/Enumerative/Partition/Basic.lean

```diff
@@ -224,6 +224,88 @@ def distincts (n : ℕ) : Finset n.Partition :=
 def oddDistincts (n : ℕ) : Finset n.Partition :=
   odds n ∩ distincts n
 
+/-- If `s ≠ 0`, then `s.sup ∈ s`. -/
+private lemma sup_mem_of_ne_zero (s : Multiset ℕ) (hs : s ≠ 0) : s.sup ∈ s := by
```

Comment:
This seems like a generally useful lemma for the API of `Multiset.sup`. Can you move it to the corresponding file, and drop the `private`?